### PR TITLE
Refactors vendor global scope procs

### PR DESCRIPTION
**Pull request by Birdtalon**
# About the pull request

Refactors awful vendor global scope procs.

Really this entire file needs looking at why do vending machines need 1300 lines. This is a start at least.

Also removed a bunch of commented out code at the bottom.

# Explain why it's good for the game
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
code: Refactors vending machine global scope procs
/:cl:

### CODE REVIEW

**Review by fira — APPROVED** [review]

File: code/game/machinery/vending/cm_vending.dm [hunk 1]
```diff
@@ -1165,58 +1163,58 @@ GLOBAL_LIST_INIT(cm_vending_gear_corresponding_types_list, list(
...
-	vendor.update_icon()
+	stat &= ~IN_USE
+	icon_state = initial(icon_state)
+	update_icon()
 
-/proc/vendor_successful_vend_one(obj/structure/machinery/cm_vending/vendor, prod_type, mob/living/carbon/human/user, turf/target_turf, insignas_override)
+/obj/structure/machinery/cm_vending/proc/vendor_successful_vend_one(prod_type, mob/living/carbon/human/user, turf/target_turf, insignas_override)
 	var/obj/item/new_item
-	var/vend_flags = vendor.vend_flags
+	var/vendor_flags = vend_flags
```
Comment (harryob) [inline code/game/machinery/vending/cm_vending.dm:1203]: we can probably just use vend_flags given this var is never modified

**Review by harryob — CHANGES_REQUESTED** [review]
thanks for this

File: code/game/machinery/vending/cm_vending.dm (hunk 1 above)
Comment (Birdtalon) [reply code/game/machinery/vending/cm_vending.dm]: Done.

File: code/game/machinery/vending/cm_vending.dm [hunk 2]
```diff
@@ -443,7 +443,7 @@ GLOBAL_LIST_EMPTY(vending_products)
 //------------TGUI PROCS---------------
 
 /obj/structure/machinery/cm_vending/ui_data(mob/user)
-	return vendor_user_ui_data(src, user)
+	return vendor_user_ui_data(user)
```
Comment (harryob) [inline code/game/machinery/vending/cm_vending.dm:446]: oh wait this proc should just return what vendor_user_ui_data and we should get rid of that proc if it's not used anywhere else

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Pull request merged**